### PR TITLE
XW-4146 | Log connection types to inform future autoplay decisions

### PR DESCRIPTION
**Pull request by JamesHawking**
## Links

* https://wikia-inc.atlassian.net/browse/XW-4146

## Description

Log the connection type via Google Analytics for mobile-wiki.

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: extensions/wikia/AssetsManager/config.php [hunk 1]
```diff
@@ -491,6 +491,7 @@
 		'//skins/oasis/js/FirefoxFindFix.js',
 		'//skins/oasis/js/tabs.js',
 		'//skins/oasis/js/Tracking.js',
+        '//skins/oasis/js/connectionType.js',
```
Comment (latata) [inline extensions/wikia/AssetsManager/config.php]: tabs

File: skins/oasis/js/connectionType.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+(function (window) {
+    'use strict';
+
+    window.Wikia = window.Wikia || {};
+
+    var connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+
+    Wikia.getConnectionType = function () {
```
Comment (latata) [inline skins/oasis/js/connectionType.js]: I think we don't need it in Wikia object. I would leave it private

File: skins/oasis/js/connectionType.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        return connection.effectiveType;
```
Comment (latata) [inline skins/oasis/js/connectionType.js]: what if connection is undefined? :)

File: skins/oasis/js/connectionType.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    if (Wikia.getConnectionType()) {
+        Wikia.Tracker.track({
+            action: Wikia.Tracker.ACTIONS.VIEW,
+            category: 'connection-type',
+            label: Wikia.getConnectionType(),
+            trackingMethod: 'analytics'
+        });
+    }
+})(window);
```
Comment (latata) [inline skins/oasis/js/connectionType.js]: new line

**Review by latata — APPROVED** [review]

File: skins/oasis/js/connectionType.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+(function (window) {
+    'use strict';
+
+    window.Wikia = window.Wikia || {};
+    var connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+
+    Wikia.getConnectionType = function () {
```
Comment (latata) [inline skins/oasis/js/connectionType.js]: I think we don't need it globally, I'd leave it private

File: skins/oasis/js/connectionType.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        return connection.effectiveType;
```
Comment (latata) [inline skins/oasis/js/connectionType.js]: What if connection is undefined?

File: skins/oasis/js/connectionType.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+(function () {
+    'use strict';
+
+    var connection = navigator.connection || navigator.mozConnection || navigator.webkitConnection;
+
+    if (connection && connection.effectiveType) {
+        Wikia.Tracker.track({
+            action: Wikia.Tracker.ACTIONS.VIEW,
+            category: 'connection-type',
+            label: Wikia.getConnectionType(),
```
Comment (bkoval) [inline skins/oasis/js/connectionType.js]: Where is this function defined? :D

Comment (latata) [inline skins/oasis/js/connectionType.js]: It should be `connection.effectiveType`

Comment (xkxd) [inline skins/oasis/js/connectionType.js]: `VM182:1 Uncaught TypeError: Wikia.getConnectionType() is not a function`

Comment (bkoval) [reply skins/oasis/js/connectionType.js]: giphy 2

Comment (JamesHawking) [reply skins/oasis/js/connectionType.js]: Forgot to change after refactor, thanks for catch :)

Comment (JamesHawking) [reply skins/oasis/js/connectionType.js]: Forgot to change after refactor, thanks for catch :)

Comment (JamesHawking) [reply skins/oasis/js/connectionType.js]: Forgot to change after refactor, thanks for catch :)

**Review by latata — APPROVED** [review]

**Review by bkoval — APPROVED** [review]
Kaboom! Good job!

**Pull request merged**